CMakeLists.txt

```diff
@@ -1,9 +1,12 @@
-cmake_minimum_required(VERSION 3.5.0 FATAL_ERROR)
-
-enable_language(Fortran)
+cmake_minimum_required(VERSION 3.5.0)
+project(stdlib Fortran)
+enable_testing()
 
-project(stdlib)
+# this avoids stdlib and projects using stdlib from having to introspect stdlib's directory structure
+set(CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_BINARY_DIR})
 
-enable_testing()
+# compiler feature checks
+include(CheckFortranSourceCompiles)
+check_fortran_source_compiles("error stop i; end" f18errorstop SRC_EXT f90)
```

Comment:
Good call here. A bunch of compilers don't support non-constant `stop` or `error stop` statements.